File: app/src/main/java/com/example/expense_management/Database/DataViewModel.java
package com.example.expense_management.Database;

import android.app.Application;

import androidx.annotation.Nullable;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;


import java.util.List;

public class DataViewModel extends AndroidViewModel {
    private Repo repository;
    private LiveData<List<DataItems>> liveDataList;

    public DataViewModel(@Nullable Application application) {
        super(application);
        this.repository=new Repo(application);
        this.liveDataList=this.repository.getDataAscending();
    }
    public LiveData<List<DataItems>> getAllDataAscending(){
        this.liveDataList=this.repository.getDataAscending();
        return liveDataList;
    }
    public LiveData<List<DataItems>> getAllDataDescending(){
        this.liveDataList=this.repository.getDataDescending();
        return liveDataList;
    }
    public LiveData<List<DataItems>> getAllExpenseDataDescending(){
        this.liveDataList=this.repository.getDataExpenseDescending();
        return liveDataList;
    }
    public LiveData<List<DataItems>> getAllGotDataDescending(){
        this.liveDataList=this.repository.getDataGotDescending();
        return liveDataList;
    }
    public LiveData<List<DataItems>> getAllExpenseDataAscending(){
        this.liveDataList=this.repository.getDataExpenseAscending();
        return liveDataList;
    }
    public LiveData<List<DataItems>> getAllGotDataAscending(){
        this.liveDataList=this.repository.getDataGotAscending();
        return liveDataList;
    }
    public void insertData(DataItems items){
        this.repository.insertData(items);
    }
    public void deleteData(DataItems items){
        this.repository.deleteData(items);
    }
    public void updateDate(long uDate,long date){
        this.repository.updateDate(uDate,date);
    }
    public void update(DataItems items){
        this.repository.update(items);
    }
    public LiveData<DataItems> getRow(long date){
       return this.repository.getRow(date);
    }
}

File: app/src/main/java/com/example/expense_management/database/DataViewModel.kt
package com.example.expense_management.database


import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData

class DataViewModel(application: Application?) : AndroidViewModel(application!!) {
    private val repository: Repo = Repo(application)
    private var liveDataList: LiveData<List<DataItems>>
    fun allDataAscending(): LiveData<List<DataItems>> {
        return repository.getDataAscending()
    }

    fun allDataDescending(): LiveData<List<DataItems>> {
        return repository.getDataDescending()
    }

    fun allExpenseDataDescending(): LiveData<List<DataItems>> {
        return repository.getDataExpenseDescending()
    }

    fun allGotDataDescending(): LiveData<List<DataItems>> {
        return repository.getDataGotDescending()
    }

    fun allExpenseDataAscending(): LiveData<List<DataItems>> {
        return repository.getDataExpenseAscending()
    }

    fun allGotDataAscending(): LiveData<List<DataItems>> {
        return repository.getDataGotAscending()
    }

    fun insertData(items: DataItems?) {
        repository.insertData(items!!)
    }

    fun deleteData(items: DataItems?) {
        repository.deleteData(items!!)
    }

    fun updateDate(uDate: Long, date: Long) {
        repository.updateDate(uDate, date)
    }

    fun update(items: DataItems?) {
        repository.update(items)
    }

    fun getRow(date: Long): LiveData<DataItems?>? {
        return repository.getRow(date)
    }

    init {
        liveDataList = repository.getDataAscending()
    }
}